src/models/neural_networks/base.py

```diff
@@ -190,6 +199,139 @@ def predict(self) -> Tuple[Dict[str, Dict[str, np.ndarray]], Dict[str, np.ndarra
 
         return test_arrays_dict, preds_dict
 
+    def unsupervised_warm_up(
+        self,
+        neighbouring_distance: Union[float, Tuple[float, float]] = 1,
+        multiplier: int = 2,
+        num_epochs: int = 1,
+        early_stopping: Optional[int] = None,
+        batch_size: int = 256,
+        learning_rate: float = 1e-3,
+        val_split: float = 0.1,
+        margin: float = 50,
+        l2: float = 0,
+    ) -> None:
+
+        if early_stopping is not None:
+            len_mask = len(
+                DataLoader._load_datasets(
+                    self.data_path,
+                    mode="train",
+                    experiment=self.experiment,
+                    shuffle_data=False,
+                    pred_months=self.pred_months,
+                )
+            )
+            train_mask, val_mask = train_val_mask(len_mask, val_split)
+
+            train_dataloader = self.get_triplet_dataloader(
+                neighbouring_distance=neighbouring_distance,
+                multiplier=multiplier,
+                mask=train_mask,
+                to_tensor=True,
+                shuffle_data=True,
+            )
+            val_dataloader = self.get_triplet_dataloader(
+                neighbouring_distance=neighbouring_distance,
+                multiplier=multiplier,
+                mask=val_mask,
+                to_tensor=True,
+                shuffle_data=True,
+            )
+
+            batches_without_improvement = 0
+            best_val_score = np.inf
+        else:
+            train_dataloader = self.get_triplet_dataloader(
+                neighbouring_distance=neighbouring_distance,
+                multiplier=multiplier,
+                to_tensor=True,
+                shuffle_data=True,
+            )
+
+        # initialize the model
+        if self.model is None:
+            x_ref, _, _ = next(iter(train_dataloader))
+            model = self._initialize_model(self._input_to_tuple(x_ref))
```

Comment:
does this train the LSTM model? Don't we need to initialise with a CNN as they use in Tile2Vec?

Reply:
The principles of tile2vec can be used with any model that takes a raw input and outputs an embedding.

So yea, in this case it can also train the (EA)LSTM model

Reply:
okay gotcha.

So have i interpreted this correctly:

"We use the unsupervised learning algorithm described in Tile2Vec to pretrain (initialise) the weights of the EALSTM. This allows us to produce weights in the network that produce sensible spatial patterns. Mainly that pixels close together are more similar than pixels that are far apart."

Reply:
yea, that's exactly right